src/Widgets/PowerModeButton.vala

```diff
@@ -19,72 +19,104 @@
+
+    private Gtk.RadioButton saver_radio;
+    private Gtk.RadioButton balanced_radio;
+    private Gtk.RadioButton performance_radio;
+
+    construct {
+        try {
+            pprofile = Bus.get_proxy_sync (BusType.SYSTEM, POWER_PROFILES_DAEMON_NAME, POWER_PROFILES_DAEMON_PATH, DBusProxyFlags.NONE);
+        } catch (Error e) {
+            critical (e.message);
+            return;
         }
 
-        private List<string> get_available_power_profiles (PowerProfile pprofile) {
-            List<string> profiles = new List<string> ();
-            for (int j = 0; j < pprofile.profiles.length; j++) {
-                profiles.append (pprofile.profiles[j].get ("Profile").get_string ());
+        var saver_icon = new Gtk.Image.from_resource ("/io/elementary/switchboard/plug/power/32x32/apps/power-mode-powersaving.svg");
+
+        var saver_label = new Gtk.Label (_("Power Saver"));
+
+        var saver_button_box = new Gtk.Box (Gtk.Orientation.VERTICAL, 3);
+        saver_button_box.pack_start (saver_icon);
+        saver_button_box.pack_end (saver_label);
+
+        saver_radio = new Gtk.RadioButton (null);
+        saver_radio.get_style_context ().add_class ("image-button");
+        saver_radio.add (saver_button_box);
+
+        var balanced_icon = new Gtk.Image.from_resource ("/io/elementary/switchboard/plug/power/32x32/apps/power-mode-balanced.svg");
+
+        var balanced_label = new Gtk.Label (_("Balanced"));
+
+        var balanced_button_box = new Gtk.Box (Gtk.Orientation.VERTICAL, 3);
+        balanced_button_box.pack_start (balanced_icon);
+        balanced_button_box.pack_end (balanced_label);
+
+        balanced_radio = new Gtk.RadioButton.from_widget (saver_radio);
+        balanced_radio.get_style_context ().add_class ("image-button");
+        balanced_radio.add (balanced_button_box);
+
+        var performance_icon = new Gtk.Image.from_resource ("/io/elementary/switchboard/plug/power/32x32/apps/power-mode-performance.svg");
+
+        var performance_label = new Gtk.Label (_("Performance"));
+
+        var performance_button_box = new Gtk.Box (Gtk.Orientation.VERTICAL, 3);
+        performance_button_box.pack_start (performance_icon);
+        performance_button_box.pack_end (performance_label);
+
+        performance_radio = new Gtk.RadioButton.from_widget (saver_radio);
+        performance_radio.get_style_context ().add_class ("image-button");
+        performance_radio.add (performance_button_box);
+
+        homogeneous = true;
+        spacing = 6;
+
+        for (int i = 0; i < pprofile.profiles.length; i++) {
+            switch (pprofile.profiles[i].get ("Profile").get_string ()) {
+                case "power-saver":
+                    add (saver_radio);
+                    break;
+                case "balanced":
+                    add (balanced_radio);
+                    break;
+                case "performance":
+                    add (performance_radio);
+                    break;
```

Comment:
Should really provide a `default:` entry as it cannot be guaranteed that the returned profile be one of those handled.